api/auth.proto

```diff
@@ -0,0 +1,46 @@
+syntax = "proto3";
+
+package envoy.api.v2;
+
+import "api/sds.proto";
+
+message AuthAction {
+  // Should we do white-list or black-list style access control.
+  enum ActionType {
+    // Request matches all rules are allowed, otherwise denied.
+    ALLOW = 0;
+    // Request matches all rules or missing required auth fields are denied,
+    // otherwise allowed.
+    DENY = 1;
+  }
+
+  ActionType action_type = 1;
+
+  // Logic AND that requires all rules match.
+  message AndRule {
+    repeated Rule rules = 1;
+  }
+
+  // Logic OR that requires at least one rule matches.
+  message OrRule {
+    repeated Rule rules = 1;
+  }
+
+  // Check peer identity using X.509 certificate.
+  message X509Rule {
```

Comment:
Do we need the indirection, i.e. can we just reference `CertificateValidationContext` below instead of `X509Rule`?

Reply:
i was trying to align with the other rule names, unfortunately there is no *typedef* in protobuf.

i could just use *CertificateValidationContext* here.